src/ImageSequencer.js

```diff
@@ -324,12 +343,38 @@ ImageSequencer = function ImageSequencer(options) {
     return this;
   }
 
+  function saveMetaModule(name, sequenceString) {
+    const sequence = stringToJSON(sequenceString);
+    // Save the given sequence string as a module
+    if (options.inBrowser) {
+      // Inside the browser we save the meta-modules using the Web Storage API
+      var metaModules = JSON.parse(window.localStorage.getItem('metaModules'));
```

Comment:
Hmm, maybe we should think about the ability to `createMetaModule()` separately from where to store it. For example, shouldn't we be able to do:

```js
var cropAndBlur = sequencer.createMetaModule('cropAndBlur', 'crop,blur');

sequencer. addSteps(cropAndBlur); // although i know we've used strings for this in the past... thoughts?
```

So a meta module would really be like a real module? Mind-bending, i know!

Reply:
Would we then essentially need to include a whole sequencer inside the meta module? so it can accept an input, and generate an output? Or is there a simpler way? 

🤯

Reply:
@jywarren according to current structure we can do sequencer.saveMetaModule('cropAndBlur','crop(),blur()')
sequencer.addSteps('cropAndBlur')

Reply:
@jywarren we don't need that, we can just register it like a dynamic module from a string

Reply:
Ah, but can't we just save it in memory for the time being, rather than in localStorage, and then think separately about reinstantiating them from other sources? I just want to think modularly here. `saveMetaModule()` could just generate the JSON, and then we could have separate methods for storing that in a persistent manner?

Reply:
that'd simplify `saveMetaModule` and externalize the function of where to do persistent storage

Reply:
maybe we'd call a separate function `storeMetaModules(...)` and `loadMetaModulesFromStorage(...)`? What do you think?

Reply:
@jywarren what do you mean by save to memory?? I was thinking we can start off with localstorage and later we can maybe shift to something like indexedDB when we add pwa elements to demo!!

Reply:
@jywarren so you want me to break the function up for now, correct? One more thing, do we pass a parameter to loadModulesfromStorage for which source we want to use?

Reply:
Like, just store it temporarily in the current instance of the sequencer. If you closed the window, you'd lose it. 

I like the idea of storing modules in localstorage or wherever, i just don't think we should merge that function with that of creating meta modules, you know? Let's do those steps separately, esp. since creation doesn't have to "know" where they're going to be stored.

Reply:
@jywarren Doing this now!!

Reply:
@jywarren But one thing is that meta-module is nothing but the json of a stringified sequence and we already have a sequencer.toJSON function which creates the json ie a meta-module

Reply:
so the function saveasMetaModule calls sequencer.convertToJSON and saves the json ie this function just does the storing part like you suggested..

Reply:
> do we pass a parameter to loadModulesfromStorage for which source we want to use

Yeah, and, i'm a bit torn about whether we ought to include module storage in this at all... it kind of goes against the idea of this being a stateless module. But I can see both ways. 

What if we focused first on the idea of meta modules as something where you could submit a dynamic module (in a separate repository) where it actually just used the meta-modules functions to define what it does internally. It'd use this code, but the means for sharing and storing such modules would be more clear. 

So, just for a thought experiment, could someone create a module in a separate repo, like:

```js
var cropAndBlur = sequencer.createMetaModule('cropAndBlur', 'crop,blur');
module.exports = [cropAndBlur, require('./info')];
```

I know I'm asking a lot here, but it helps illustrate the kind of compartmentalization that would be really useful. What do you think?

Reply:
I see what you're saying about `convertToJSON`. Does my example of a dynamic module above help illustrate what I'm talking about?

Reply:
@jywarren yeah I do get what you mean. actually the thing is I thought of meta-modules as macros or just a sequence of steps but I am getting you want these to be complete modules in themselves, are you sure about this??

Reply:
Right now meta-modules cannot be made in a separate repository since they are nothing but the sequence in json, i can add a function that will make what you say possible!!

Reply:
Well, if `createMetaModule` accepts a string and returns a fully functional module, this would be possible, right? I think that although this is complex, it is worthwhile. How long do you think it'd take to implement?

Reply:
Yes, true modules that are made up of other modules!

Reply:
And thank you for thinking through this with me!

Reply:
@jywarren Actually the thing is that right now we are pulling the info in from the info of constituent modules so in order to have it support info for meta modules and have it load like normal modules needs fundamentally changing the way we do modules and bake in meta-modules there

Reply:
Actually I think we are better off this way coz even if we try to make meta-modules like fully functional modules at some point we would have to expand them, what i mean is that at some point we would have to remove the meta-module from our steps array and we would have to push in the constituent modules before calling `sequencer.run()`

Reply:
@jywarren This is a little hard to explain, Actually I started doing it but I realized this that we cannot make them like normal modules unless we find some way to execute the steps inside the meta-module like you said have an instance of sequencer inside of the module to execute the sequence and return the output